app/controllers/thredded/posts_controller.rb

```diff
@@ -27,6 +27,12 @@ def topic
       post.postable
     end
 
+    def destroy
+      post.destroy!
+
+      redirect_to request.referrer
```

Comment:
Rails 5 defines a new redirect_back method that requires a mandatory `fallback_location` argument. We can use it even on Rails 4, because we polyfill it in the ApplicationController.